packages/connection/src/class-manager.php

```diff
@@ -829,6 +829,10 @@ public function register( $api_endpoint = 'register' ) {
 		add_action( 'pre_update_jetpack_option_register', array( '\\Jetpack_Options', 'delete_option' ) );
 		$secrets = $this->generate_secrets( 'register', get_current_user_id(), 600 );
 
+		if ( false === $secrets ) {
+			return new WP_Error( 'cannot_save_secrets', __( 'Could not save secrets. Please confirm that the options table is writable.', 'jetpack' ) );
```

Comment:
I wonder if the message could be more user-friendly, and suggest a possible solution to the site owner? Maybe something like "We suggest that you contact your hosting provider, and ask them for help checking that the options table is writable on your site"?

Reply:
Done, please re-review 👍